src/Plate/coordinate.ts

```diff
@@ -0,0 +1,131 @@
+import { CoordinateSystem } from './coordinateSystem';
+import { Coordinates, FlowDirection } from './types';
+
+export class Coordinate {
+  public row: Coordinates['row'];
+
+  public column: Coordinates['column'];
+
+  public coordinateSystem: CoordinateSystem;
+
+  constructor(
+    row: Coordinates['row'],
+    column: Coordinates['column'],
+    coordinateSystem: CoordinateSystem,
+  ) {
+    if (!coordinateSystem.rows().includes(row)) {
+      throw new Error(
+        `Expected a row with value of ${coordinateSystem
+          .rows()
+          .join(',')}, got ${row}.`,
+      );
+    }
+    this.row = row;
+
+    if (!coordinateSystem.columns().includes(column)) {
+      throw new Error(
+        `Expected a column with value of ${coordinateSystem
+          .columns()
+          .join(',')}, got ${column}.`,
+      );
+    }
+    this.column = column;
+
+    this.coordinateSystem = coordinateSystem;
+  }
+
+  static fromString(
+    coordinateString: string,
+    coordinateSystem: CoordinateSystem,
+  ): Coordinate {
+    const rows = coordinateSystem.rows();
+    const rowsOptions = rows.join('|');
+
+    const columns = [
+      ...coordinateSystem.columns(),
+      ...coordinateSystem.paddedColumns(),
+    ];
+    const columnsOptions = columns.join('|');
+
+    const matches = coordinateString.match(
+      new RegExp(`^(${rowsOptions})(${columnsOptions})`),
+    );
+
+    if (matches === null || matches?.length === 0) {
+      const firstValidExample = rows[0].toString() + columns[0].toString();
+      const lastValidExample =
+        rows[rows.length - 1].toString() +
+        columns[columns.length - 1].toString();
+      const coordinateSystemClass = coordinateSystem.constructor.name;
+      throw new Error(
+        `Expected a coordinate between ${firstValidExample} and ${lastValidExample} for ${coordinateSystemClass}, got: ${coordinateString}.`,
+      );
+    }
+
+    return new this(
+      matches[1] as Coordinates['row'],
+      Number(matches[2]) as Coordinates['column'],
```

Comment:
Possibly undesired results here as well -> explicit checks for the array length + tests with empty parameters would improve robustness

![image](https://user-images.githubusercontent.com/4804412/161546820-e80b4160-0e24-437c-9718-0af12831fb98.png)